src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -188,7 +188,6 @@ export class DatasetService {
           ? ({} as IndexPatternFieldMap)
           : await type?.fetchFields(dataset, services);
         const spec = {
-          id: dataset.id,
```

Comment:
Do we need to prefixed the id with data source id? I recall dataset has a special logic to determine if there is data source associated from dataset id(${dataSourceId}::${indexPatternId})